agent/config/agentconfig.go

```diff
@@ -106,12 +112,18 @@ type AgentConfigWriter interface {
 
 // ReadAgentConfig is a helper to read either machine or controller agent config,
 // whichever is there. Machine config gets precedence.
-func ReadAgentConfig(c AgentConfigWriter, agentId string) error {
+func ReadAgentConfig(c AgentConfigReader, agentId string) error {
 	var tag names.Tag = names.NewMachineTag(agentId)
-	err := c.ReadConfig(tag.String())
-	if err != nil {
-		tag = names.NewControllerAgentTag(agentId)
-		err = c.ReadConfig(tag.String())
+	machineErr := c.ReadConfig(tag.String())
+	if machineErr == nil {
+		return nil
 	}
-	return errors.Trace(err)
+
+	tag = names.NewControllerAgentTag(agentId)
+	ctrlErr := c.ReadConfig(tag.String())
+	if ctrlErr == nil {
+		return nil
+	}
+
+	return fmt.Errorf("reading agent config for %q: %w, %w", agentId, machineErr, ctrlErr)
```

Comment:
You could use `errors.Join` here but it will add a new line between the errors. Might not be ideal.

Reply:
Yeah, non-ideal, although it could be workable.